### DISCUSSION TITLE
Implement Discord Bot Integration for Automatic GitHub Notifications

### DISCUSSION THREAD

**Discussion by Niravanaa** [NathanGrenier/SOEN-341#14] [Ideas]
Introducing a Discord bot integration for our repository to streamline notifications. By setting up a webhook from our GitHub repository to Discord, we can receive instant updates on all events directly in our Discord server, that we primarily use for short/quick discussion.

This would make sure none of us are left behind in knowing the latest updates in the repo.

**Comment by NathanGrenier** [comment]
Could be useful. Usually I just use the emails that github sends me. 

If you want you can create an issue from this discussion and set it up. It would be nice to get notifications for these events:
- New issue is created
- New discussion is created
- New PRs

**Reply by Niravanaa** [reply]
Here are all the notifications we can get from a Webhook on our Discord Server, besides the events that you listed, should we also consider getting notified for any other of these events?

![image](https://github.com/NathanGrenier/SOEN-341/assets/64038962/575b7eec-5584-41d9-8deb-eafd25727d9d)
![image](https://github.com/NathanGrenier/SOEN-341/assets/64038962/13f25867-c9b9-416f-862f-8501ef35b7a4)
![image](https://github.com/NathanGrenier/SOEN-341/assets/64038962/bf052634-cdf3-45b3-b09d-4fbb68a55a03)

**Reply by NathanGrenier** [reply]
I looked through the options and listed which ones I think would be useful. It would be cool to have separate channels dedicated to certain notifications.

Here's what I think would be useful. The names in bold would correspond to the discord channel that the notifications would be in.

**Documentation:**
- Wiki

**Deployments:**
- Deployments
- Releases

**Discussions:**
- Discussions 
- Discussion Comments

**Issues:**
- Issue Comments
- Issues

**PR:**
- Workflow runs
- Pull request reviews
- Pushes (Only for certain branches [main, dev])
